### ISSUE TITLE
firejail.wrappedBinaries has no effect

### ISSUE THREAD

**Issue by aidalgol** [NixOS/nixpkgs#152852]
### Describe the bug
`programs.firejail.wrappedBinaries` does not appear to have any effect on whether programs are run under `firejail`.

### Steps To Reproduce
1. Add the following to `/etc/nixos/configuration.nix`
  ```nix
  programs.firejail.enable = true;
  programs.firejail.wrappedBinaries = {
    thunderbird = {
      executable = "${lib.getBin pkgs.thunderbird}/bin/thunderbird";
      profile = "${pkgs.firejail}/etc/firejail/thunderbird.profile";
    };
  };
  ```
2. Run `sudo nixos-rebuild switch`
3. In a separate terminal, run `thunderbird`
4. Run `firejail --list`
    See that no sandboxes are listed, even though `thunderbird` should have been run under `firejail`.

### Expected behavior
I expected the command `thunderbird` to run the Thunderbird application through `firejail`, as described in the [option description](https://search.nixos.org/options?channel=21.11&show=programs.firejail.wrappedBinaries&from=0&size=50&sort=relevance&type=packages&query=firejail) for `programs.firejail.wrappedBinaries`.

### Notify maintainers
@peterhoeg 

### Metadata
```console
$ nix-shell -p nix-info --run "nix-info -m"
these paths will be fetched (0.07 MiB download, 0.34 MiB unpacked):
  /nix/store/gd7ihraqqpginv5cmhgk347nfn2iiqkj-bash-interactive-5.1-p8-dev
  /nix/store/k8qjik2a44ljdx48lmin2byz77ssrngk-stdenv-linux
copying path '/nix/store/gd7ihraqqpginv5cmhgk347nfn2iiqkj-bash-interactive-5.1-p8-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/k8qjik2a44ljdx48lmin2byz77ssrngk-stdenv-linux' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.88, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11.334921.8588b14a397"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

**Comment by hqurve** [comment]
From https://firejail.wordpress.com/documentation-2/firefox-guide/

> by default, a single Firefox process instance handles multiple browser windows. If you already have Firefox running, you would need to use -no-remote command line option, otherwise you end up with a new tab or a new window attached to the existing Firefox process

I believe the same is true for thunderbird. That is, if I already have thunderbird open and then run `firejail thunderbird` (which is essentially all that option does https://github.com/NixOS/nixpkgs/blob/2a2ffb13517c34f6c74ef0a18746f546a1e7d0e9/nixos/modules/programs/firejail.nix#L27), a new tab in the already running thunderbird window opens and the the firejail process ends.

However, if I have no instance of thunderbird running, thunderbird stays in firejail as exepected.

**Comment by aidalgol** [comment]
I did not have thunderbird running when I followed the steps outlined in this issue.

**Comment by hqurve** [comment]
Does thunderbird start in firejail when manually invoking `firejail thunderbird` (or equivalent). Also, what is the result of `cat $(which thunderbird)`?

**Comment by aidalgol** [comment]
```console
$ cat $(which thunderbird)
#! /nix/store/v1d8l3zqnia3hccqd0701szhlx22g54z-bash-5.1-p8/bin/bash -e
export LD_LIBRARY_PATH='/nix/store/8s9kjhmsyqzsja20bh2z9nmnxjd81zzd-gpgme-1.16.0/lib'${LD_LIBRARY_PATH:+':'}$LD_LIBRARY_PATH
export PATH='/nix/store/k2pfqhw31c3hr04riiz6as01xcqfs749-gnupg-2.2.27/bin'${PATH:+':'}$PATH
exec -a "$0" "/nix/store/g7ba417hyd0cqkdy8w2dg7p1b9hki593-thunderbird-91.4.0/bin/.thunderbird-wrapped_"  "$@"
```
```console
$ readlink -f $(which thunderbird)
/nix/store/g7ba417hyd0cqkdy8w2dg7p1b9hki593-thunderbird-91.4.0/bin/thunderbird
```

Explicitly invoking `firejail thunderbird` behaves as expected (thunderbird is run in a firejail sandbox).

**Comment by hqurve** [comment]
For some reason that is the normal thunderbird, not the one created by the firejail module. Out of curiousity, do you also have thunderbird in `environment.systemPackages`?

**Comment by aidalgol** [comment]
Yes I do.  Is that going to cause conflicts?

**Comment by hqurve** [comment]
Yes... I am unsure what the order is in general (when priorities are the same), but if you have two derivations with the same file name, only one persists when linked. So, in this case it seems that the executable thunderbird took priority while linking the systempath and your firejail version got overwritten.

Also, the systemPackages option ignores collisions :p https://github.com/NixOS/nixpkgs/blob/5b091d4fbe3b7b7493c3b46fe0842e4b30ea24b3/nixos/modules/config/system-path.nix#L172

See: https://github.com/NixOS/nixpkgs/blob/5b091d4fbe3b7b7493c3b46fe0842e4b30ea24b3/pkgs/build-support/buildenv/default.nix

**Comment by aidalgol** [comment]
I actually only put it in `environment.systemPackages` because I did not realise that the `firejail.wrappedBinaries` option would work without specifying the package there.  I am running a bare window manager, and almost completely ignore `.desktop` files, so the solution for me is to simply remove it from `systemPackages`.

I have submitted a pull request that tries to clarify the description of `firejail.wrappedBinaries`, and I consider this issue resolved.  Thank you very much for your help here!
